docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,406 @@
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+###### From only policy (policies that do not have to)
+
+Policies:
+- MeshTrafficPermission
+- MeshRateLimit
+- MeshGlobalRateLimit
+- MeshFaultInjection
+
+```yaml
+kind: MeshTrafficPermission
+spec:
+  targetRef:
+    kind: MeshExternalService
+    name: httpbin
+    proxyType: ["egress"]
+  from:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      default:
+        action: Allow
+    - targetRef:
+        kind: Mesh
+      default:
+        action: Deny
+```
+
+###### `To` policy on sidecar
+
+Policies:
+- MeshRetry
+- MeshHTTPRoute
+- MeshTCPRoute
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshRetry
+metadata:
+  name: backend
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: ["sidecar"]
+  to:
+    - targetRef:
+        kind: MeshExternalService
+        name: backend
+      default:
+        http:
+          numRetries: 10
+```
+
+###### `To` policy on egress
+
+The following policy we're going to configure on Egerss: 
+- `MeshCircuitBreaker` - we don't want to cut out whole egress from traffic only faulty instance of MeshExternalService
+- `MeshLoadBalancingStrategy` - (we only care about `default.loadBalancer` portion) we don't want to change the loadbalancing to Egress but to the underlying MeshExternalServices
+- `MeshHealthCheck` - we don't want to be health checking Egress but the underlying MeshExternalServices
+
+```yaml
+type: MeshCircuitBreaker
+mesh: default
+name: circuit-breaker
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: ["egress"]
+  to:
+    - targetRef:
+        kind: MeshExternalService
+        name: backend
+      default:
+        outlierDetection:
+          detectors:
+            totalFailures:
+              consecutive: 10
+```
+
+###### `To` and `From` policies
+
+Policies:
+- MeshAccessLog
+- MeshTimeout
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: ["egress"]
+  to:
+  - targetRef:
+      kind: MeshExternalService
+      name: backend
+    default: ...
+```
+
+##### Empty `proxyType`
+
+Currently, egress is not selected by `kind: Mesh` targetRef, but missing `proxyType` means both `gateway` and `sidecar`.
+We don't want to assume empty `proxyType` selecting `egress` because we want greater control over what is configured and to make sure user is aware that egress is affected.
+To avoid breaking changes now and in the future we can extend this more easily.
+
+We will require `proxyType` in validation whenever `MeshExternalService` is referenced.
+
+#### Only on the egress
+
+There are policies that make sense to be in both places and the ones like MeshTimeout that only have drawbacks on egress.
+
+#### Only on sidecar
+
+There are policies that make more sense on egress and in both places.
```

Comment:
What do these lines mean?

Reply:
Just wanted to document that we considered configuring everything on egress or on sidecar.

Reply:
Ah ok then maybe like

```suggestion
#### Only configuring policies on the egress

This doesn't work since there are policies that make sense to be in both places and the ones like MeshTimeout that only have drawbacks on egress.

#### Only configuring policies on sidecar

This doesn't work because there are policies that make more sense on egress and in both places.
```

because as is it just reads like a non sequitur